src/environment.cc

```diff
@@ -179,6 +180,29 @@ const std::string& Environment::KubernetesClusterName() const {
   return kubernetes_cluster_name_;
 }
 
+const std::string& Environment::KubernetesClusterLocation() const {
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (kubernetes_cluster_location_.empty()) {
+    if (!config_.KubernetesClusterLocation().empty()) {
+      kubernetes_cluster_location_ = config_.KubernetesClusterLocation();
+    } else {
+      // Get the kube-env.
+      const std::string kube_env =
+          GetMetadataString("instance/attributes/kube-env");
+      // kube-env is a list of NAME: VALUE pairs, one per line.
+      // The actual location is in the ZONE variable.
+      // TODO: Refactor this into GetKubeEnv("ZONE") in kubernetes.cc.
+      std::istringstream in(kube_env);
+      for (std::string line; std::getline(in, line); ) {
+        if (line.find("ZONE: ") == 0) {
+          kubernetes_cluster_location_ = line.substr(line.find(':') + 2);
```

Comment:
Let's break out of this early when we find a match.

Reply:
Obsolete. See the new shiny in the next push.